kiwi/bootloader/config/grub2.py

```diff
@@ -1146,6 +1146,11 @@ def _fix_grub_root_device_reference(self, config_file, boot_options):
                         'root={0}'.format(boot_options.get('root_device')),
                         self.root_reference
                     )
+                    grub_config = re.sub(
+                        r'root=PARTUUID=[a-z0-9-]+',
+                        r'{0}'.format(self.root_reference),
+                        grub_config
+                    )
```

Comment:
Rather than increasing the cases we try to guess it I am wondering if it wouldn't be better to apply one of the two ideas I had:

1. Always apply our `root=` for the OBS and/or overlay builds
2. Set some strategy to discover the write partition in https://github.com/OSInside/kiwi/blob/6cdc6bbdd350120d6984e89f058afaa160e6d72d/dracut/modules.d/90kiwi-overlay/kiwi-overlay-root.sh#L23 and forget about having to parse `root=overlay:UUID`. That would make overlay module transparent to grub2-mkconfig

Reply:
I also thought about always replacing root= with the correct value when we run in obs. That will make the code easier but we will then always patch the mkconfig result... which could live with to be honest

The second approach I'm not sure about. The reason for root=overlay= is because there is a dracut module that handles this special setup. So we need a trigger for it and as it works nicely I don't think the lack of obs-worker capabilities servers the purpose for a design change

If ok I would go with 1)

Thoughts ?

PS:

This patch here worked, for me. The current overlay based images in the integration tests now boots again. Could double check on them too and maybe give a standard image (no overlay) also a short test ?

Thanks

Reply:
Also there is another alternative which a is a variant of 1). What about setting always the `root=` for the overlay case (within or not OBS) and then keep the OBS case exactly as it was. I'd say this is not very intrusive and also opens the door to the possibility of not touching the configuration inside OBS if unneeded (probably this is already the case in TW :man_shrugging: )

In any case I am fine with 1)

Reply:
> This patch here worked, for me. The current overlay based images in the integration tests now boots again. Could double check on them too and maybe give a standard image (no overlay) also a short test ? 

I am giving a quick run

Reply:
JFYI I did few boots of overlay VM and regular VM disks and they all successfully booted.

Reply:
Thanks much. I think I will think about our options a bit more and let this one rest until Monday (or maybe the weekend). Thanks for all your feedback, much appreciated.